Examples/Io/Performance/ActsExamples/Io/Performance/VertexPerformanceWriter.cpp

```diff
@@ -402,89 +461,220 @@ ActsExamples::ProcessCode ActsExamples::VertexPerformanceWriter::writeT(
+                m_recoQOverP.push_back(recoMom[2]);
+
+                Acts::ActsVector<3> diffMom = recoMom - trueMom;
+                // Accounting for the periodicity of phi. We overwrite the
+                // previously computed value for better readability.
+                diffMom[0] = Acts::detail::difference_periodic(
+                    recoMom(0), trueMom(0), 2 * M_PI);
+                m_resPhi.push_back(diffMom[0]);
+                m_resTheta.push_back(diffMom[1]);
+                m_resQOverP.push_back(diffMom[2]);
+
+                m_pullPhi.push_back(
+                    pull(diffMom[0], momCov(0, 0), "phi", false));
+                m_pullTheta.push_back(
+                    pull(diffMom[1], momCov(1, 1), "theta", false));
+                m_pullQOverP.push_back(
+                    pull(diffMom[2], momCov(2, 2), "q/p", false));
+
+                const auto& recoUnitDir = paramsAtVtx->unitDirection();
+                double overlap = trueUnitDir.dot(recoUnitDir);
+                m_momOverlap.push_back(overlap);
+              }
+
+              // Save track parameters after the vertex fit
+              const auto paramsAtVtxFitted = propagateToVtx(trk.fittedParams);
+              if (paramsAtVtxFitted != std::nullopt) {
+                Acts::ActsVector<3> recoMomFitted =
+                    paramsAtVtxFitted->parameters().segment(Acts::eBoundPhi, 3);
+                const Acts::ActsMatrix<3, 3>& momCovFitted =
+                    paramsAtVtxFitted->covariance()->block<3, 3>(
+                        Acts::eBoundPhi, Acts::eBoundPhi);
+                m_recoPhiFitted.push_back(recoMomFitted[0]);
+                m_recoThetaFitted.push_back(recoMomFitted[1]);
+                m_recoQOverPFitted.push_back(recoMomFitted[2]);
+
+                Acts::ActsVector<3> diffMomFitted = recoMomFitted - trueMom;
+                // Accounting for the periodicity of phi. We overwrite the
+                // previously computed value for better readability.
+                diffMomFitted[0] = Acts::detail::difference_periodic(
+                    recoMomFitted(0), trueMom(0), 2 * M_PI);
+                m_resPhiFitted.push_back(diffMomFitted[0]);
+                m_resThetaFitted.push_back(diffMomFitted[1]);
+                m_resQOverPFitted.push_back(diffMomFitted[2]);
+
+                m_pullPhiFitted.push_back(
+                    pull(diffMomFitted[0], momCovFitted(0, 0), "phi"));
+                m_pullThetaFitted.push_back(
+                    pull(diffMomFitted[1], momCovFitted(1, 1), "theta"));
+                m_pullQOverPFitted.push_back(
+                    pull(diffMomFitted[2], momCovFitted(2, 2), "q/p"));
+
+                const auto& recoUnitDirFitted =
+                    paramsAtVtxFitted->unitDirection();
+                double overlapFitted = trueUnitDir.dot(recoUnitDirFitted);
+                m_momOverlapFitted.push_back(overlapFitted);
+              }
+            }
+          }
```

Comment:
Maybe worth factoring out into a separate function?

Reply:
I could do this but the I couldn't use the pull lambda I guess.. Or should I factor this one out as well?

Reply:
Not super important. Just saw a big intended piece of code, that's all.